src/tests/Directory.Build.props

```diff
@@ -223,7 +223,8 @@
   <PropertyGroup Condition="'$(RuntimeFlavor)' == 'coreclr' and '$(TargetsAppleMobile)' == 'true'">
     <_targetOS>$(TargetOS)</_targetOS>
     <_targetArchitecture>$(TargetArchitecture)</_targetArchitecture>
-    <UseNativeAOTRuntime>true</UseNativeAOTRuntime>
+    <AppleAppBuilderRuntime Condition="'$(TestBuildMode)' == 'nativeaot'">NativeAOT</AppleAppBuilderRuntime>
+    <AppleAppBuilderRuntime Condition="'$(TestBuildMode)' != 'nativeaot'">CoreCLR</AppleAppBuilderRuntime>
```

Comment:
While it is fine to use: `Runtime` + `targetRuntime` variables instead of `UseNativeAOTRuntime` inside of the AppleAppBuilder, I am a little bit worried about changes around `UseNativeAOTRuntime` as an MSBuild property. /cc: @kotlarmilos